src/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -173,21 +175,24 @@ public static void CreateErrorResponse(HttpResponse response, RequestContext req
 
             try
             {
-                var msalTokenResponse = JsonHelper.DeserializeFromJson<MsalTokenResponse>(response.Body);
+                MsalTokenResponse msalTokenResponse = JsonHelper.DeserializeFromJson<MsalTokenResponse>(response.Body);
 
                 if (CoreErrorCodes.InvalidGrantError.Equals(msalTokenResponse.Error, StringComparison.OrdinalIgnoreCase))
                 {
                     throw MsalExceptionFactory.GetUiRequiredException(
                         CoreErrorCodes.InvalidGrantError,
                         msalTokenResponse.ErrorDescription,
                         null,
-                        ExceptionDetail.FromHttpResponse(response));
+                        response, 
+                        msalTokenResponse);
                 }
 
                 serviceEx = MsalExceptionFactory.GetServiceException(
```

Comment:
Seems like the api in the factory is not aligned between GetServiceException and GetUIRequiredException (parameters seems to be switched around)